src/blockdata/transaction.rs

```diff
@@ -825,19 +826,27 @@ impl crate::consensus::encode::Encodable for ExtraField {
 }
 
 impl Decodable for SubField {
-    fn consensus_decode<R: io::Read + ?Sized>(r: &mut R) -> Result<SubField, encode::Error> {
+    fn consensus_decode<R: io::Read + ?Sized + Seek>(r: &mut R) -> Result<SubField, encode::Error> {
         let tag: u8 = Decodable::consensus_decode(r)?;
 
         match tag {
             0x0 => {
                 let mut i = 0;
-                for _ in 1..u8::MAX {
-                    // Consume all bytes until the end of cursor or until 255 bytes have
+                for _ in 1..=u8::MAX {
+                    // Consume all valid padding bytes until the end of cursor or until 255 valid bytes have
                     // been consumed. A new cursor must be created when parsing extra bytes
                     // otherwise transaction bytes will be consumed
                     let byte: Result<u8, encode::Error> = Decodable::consensus_decode(r);
                     match byte {
-                        Ok(_) => {
+                        Ok(val) => {
+                            // Anything else than '0' is not a padding byte
+                            if val != 0 {
+                                // Rewind the cursor one position back to not consume the non-padding byte
+                                r.seek(io::SeekFrom::Current(-1))?;
+                                // Notify the caller that parsing failed
+                                println!("Error::ParseFailed(Invalid padding byte)");
+                                return Err(encode::Error::ParseFailed("Invalid padding byte"));
+                            }
```

Comment:
This isn't what monero-core does (rewinding the cursor), I think we should just return an error. This also means we don't need the `Seek` trait.